### ISSUE TITLE
False positive for no-unused-declaration with typescript-eslint 4.0

### ISSUE THREAD

**Issue by jsaguet** [cartant/eslint-plugin-etc#19]
I encountered a false positive with the rule `no-unused-declaration` after updating to [@typescript-eslint/parser 4.0.0](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0)
The issue happens when I type a callback function which contains arguments:

```typescript
export const myFunction = (
    args: any,
    cb: (arg: string) => void,
          ~~~                       "arg" is not used; unused declarations are forbidden.
): void => cb('value');
```
I guess the expected behavior should be as it was with older version of typescript-eslint: no unused error for typed functions arguments

**Comment by cartant** [comment]
Version 4.0.0 of the parser reimplements the ESLint scope functionality. And it's done in a manner that effects problems for this rule. Furthermore, it's difficult to see how the changes that have been made can be used and the `no-unused-vars` rule that was added - to take advantage of the scope changes - has a large number of open issues for false positives.

I've updated the rules implementation to ignore TypeScript-specific portions of the scope, but that has the effect of disabling checking for a significant number of situations. It is what it is. Until the false positives in the with the `no-unused-vars` rule and any issues with the scope implementation are addressed, I won't be spending any time trying to restore the rule's pre-4.0.0 functionality.

**Comment by cartant** [comment]
Basically, I won't be touching this until these have been investigated/addressed: https://github.com/typescript-eslint/typescript-eslint/milestone/7

**Comment by cartant** [comment]
I'm going to be deleting the `no-unused-declaration` rule. Now that the official TypeScript ESLint plugin has an implementation of [`no-unused-vars`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md), `no-unused-declaration` is pretty much redundant and I would prefer not to support it. If anyone needs to differentiate between vars and imports, the [`eslint-plugin-unused-imports`](https://github.com/sweepline/eslint-plugin-unused-imports) plugin includes `no-unused-vars-ts` and `no-unused-imports-t` rules - and the latter has a fixer.
